src/reporting/clients/MappingsClient.ts

```diff
@@ -0,0 +1,641 @@
+  public async getMappings(accessToken: AccessToken, iModelId: string, top?: number): Promise<Mapping[]> {
+    const mappings: Array<Mapping> = [];
+    const mapIterator = this.getMappingsIterator(accessToken, iModelId, top);
+    for await(const map of mapIterator) {
+      mappings.push(map);
+    }
+    return mappings;
+  }
+
+  public getMappingsIterator(accessToken: AccessToken, iModelId: string, top?: number): EntityListIterator<Mapping> {
+    if(!this.topIsValid(top)) {
+      throw new RequiredError(
+        "top",
+        "Parameter top was outside of the valid range [1-1000]."
+      );
+    }
+    let url = `${this.basePath}/datasources/imodels/${encodeURIComponent(iModelId)}/mappings`;
+    url += top ? `/?$top=${top}` : "";
+    const request = this.createRequest("GET", accessToken);
+    return new EntityListIteratorImpl(async () => getEntityCollectionPage<Mapping>(
+      url,
+      async (nextUrl: string): Promise<Collection<Mapping>> => {
+        const response: MappingCollection = await this.fetchJSON<MappingCollection>(nextUrl, request);
+        return {
+          values: response.mappings,
+          // eslint-disable-next-line @typescript-eslint/naming-convention
+          _links: response._links,
+        };
+      }));
+  }
+
+  public async getMapping(accessToken: AccessToken, iModelId: string, mappingId: string): Promise<Mapping> {
+    const url = `${this.basePath}/datasources/imodels/${encodeURIComponent(iModelId)}/mappings/${encodeURIComponent(mappingId)}`;
+    const requestOptions: RequestInit = this.createRequest("GET", accessToken);
+    return (await this.fetchJSON<MappingSingle>(url, requestOptions)).mapping;
+  }
+
+  public async createMapping(
+    accessToken: AccessToken,
+    iModelId: string,
+    mapping: MappingCreate
+  ): Promise<Mapping> {
+    if (!this.isSimpleIdentifier(mapping.mappingName)) {
+      throw new RequiredError(
+        "mappingName",
+        "Required field mappingName of mapping was missing or invalid.",
+      );
+    }
+
+    const url = `${this.basePath}/datasources/imodels/${encodeURIComponent(iModelId)}/mappings`;
+    const requestOptions: RequestInit = this.createRequest("POST", accessToken, JSON.stringify(mapping));
+    return (await this.fetchJSON<MappingSingle>(url, requestOptions)).mapping;
+  }
+
+  public async updateMapping(
+    accessToken: AccessToken,
+    iModelId: string,
+    mappingId: string,
+    mapping: MappingUpdate
+  ): Promise<Mapping> {
+    if (mapping.description == null && mapping.extractionEnabled == null && mapping.mappingName == null) {
+      throw new RequiredError(
+        "mapping",
+        "All properties of mapping were missing.",
+      );
+    }
+    if (mapping.mappingName != null && !this.isSimpleIdentifier(mapping.mappingName)) {
+      throw new RequiredError(
+        "mappingName",
+        "Required field mappingName of mapping was invalid.",
```

Comment:
```suggestion
        "Required field mappingName was invalid.",
```

Reply:
Do similar changes to all validation messages.

Reply:
applied